pkg/trace/config/apply.go

```diff
@@ -357,7 +365,39 @@ func (c *AgentConfig) applyDatadogConfig() error {
 		GRPCPort:        grpcPort,
 		MaxRequestBytes: c.MaxRequestBytes,
 	}
+	if c.TelemetryConfig == nil {
+		c.TelemetryConfig = &TelemetryConfig{}
+	}
 
+	if len(c.TelemetryConfig.Endpoints) == 0 {
+		c.TelemetryConfig.Endpoints = []*Endpoint{{
+			Host: telemetryEndpointPrefix + config.DefaultSite,
+		}}
+	}
+
+	if config.Datadog.GetBool("apm_config.telemetry.enabled") {
+		endpoints := []*Endpoint{{
+			Host:   config.GetMainEndpoint(telemetryEndpointPrefix, "apm_config.telemetry.dd_url"),
+			APIKey: c.Endpoints[0].APIKey,
+		}}
+		endpoints = appendAdditionalEndpoints(endpoints, "apm_config.telemetry.additional_endpoints")
+		// convert any URLs into Hostnames
```

Comment:
Shouldn't we use URLs here too? To be consistent. I'd expect all `additional_endpoints` functionality to work the same everywhere. It's used in many places in the agent (even outside APM). Doing it differently here could create confusion. Am I missing something?

Reply:
IMO There are two problems existing with Endpoint.Host representation. 
1) Its named Host, but holds a URL :) 
2) it holds a string, that then needs to be parsed into URL representation. If it is malformed, it will cause an error to show up later in the stack. 

I think any URL representation errors should be surfaced at the config parsing time, the same way as we check if APIKeys exist for defined additional endpoints.

Reply:
I can change it to only validate its a valid url, and not change the value here.
Though I'll have to parse it another place in the code again.

Reply:
Done, I've moved the conversion to the telemetry initialization. But kept the validation. @gbbr  Let me know what you think

Reply:
Ok per discussion on slack. I removed any validation from applyConfig. And will only rely on validation in telemetry proxy component. 
I've added a few new tests there to cover this behaviour, and cleaned up the test setup a little to make it easier on the eyes (at least mine :) )